tests/test_search_optimizer.c

```
@@ -0,0 +1,164 @@
+
+    opt = sr_optimizer_by_name("ga");
+    CLEED_TEST_ASSERT(opt != NULL);
+    CLEED_TEST_ASSERT(opt->type == SR_GENETIC);
+
+    opt = sr_optimizer_by_name("unknown");
+    CLEED_TEST_ASSERT(opt == NULL);
+
+    return 0;
+}
+
+static int test_config_from_env(void)
```

Clang (14.0.6)

[warning] function 'test_config_from_env' has cognitive complexity of 27 (threshold 25)

(readability-function-cognitive-complexity)